frame/contracts/src/benchmarking/mod.rs

```diff
@@ -270,6 +303,196 @@ benchmarks! {
 		m.step();
 	}
 
+	// This benchmarks the v14 migration step (Move code owners' reserved balance to be held instead).
+	#[pov_mode = Measured]
+	v14_migration_step {
+		mod mock {
+			use frame_support::traits::{BalanceStatus, Currency, ExistenceRequirement, Imbalance, ReservableCurrency,
+				SameOrOther, SignedImbalance, TryDrop, WithdrawReasons};
+			use sp_runtime::{DispatchResult, Saturating, DispatchError};
+			use sp_std::ops::Div;
+
+			#[derive(Default)]
+			pub struct MockBalance;
```

Comment:
Why do you need the `MockBalance`? Can't you use `pallet_balances::Pallet` here?

Reply:
`BalanceOf<T>` wouldn't satisfy `BalanceOf<T>: From<OldCurrency::Balance>` for `pallet_balances::Pallet<T>` and it isn't possible to implement the `From` trait for it

Reply:
You can add this bound to the `benchmarks!` macro. Then it should work. The thing is that it is implemented. You just need to bind it.